hudi-client/src/test/java/org/apache/hudi/table/TestMergeOnReadTable.java

```diff
@@ -1311,4 +1383,111 @@ private void assertNoWriteErrors(List<WriteStatus> statuses) {
       assertFalse("Errors found in write of " + status.getFileId(), status.hasErrors());
     }
   }
+  
+  private FileStatus[] insertAndGetFilePaths(List<HoodieRecord> records, HoodieWriteClient client,
```

Comment:
I am surprised by the amount of test helper code you needed to write yourself.. Should we be using code from some existing helpers.. or should these helpers move somewhere else, so they are beneficial over all?

Reply:
This one is refactored from existing test 'TestMergeOnReadTable#testSimpleInsertAndUpdate' . I don't know this setup is needed in other places. If you have any suggestions for new location, let me know.

There are also no tests so far with end-to-end testing of incremental reads on MOR tables (as far as i could tell). So I had to add some more utility test methods for that. Please suggest if ythere is a better location for that

Reply:
let's have it like this for now.. Thanks for the explanation..!